### ISSUE TITLE
esp32: network devices pull in GNRC dependencies statically

### ISSUE THREAD

**Issue by miri64** [RIOT-OS/RIOT#12964]
#### Description
While compile testing #12931 we noticed that the ESP32 won't build `tests/lwip` anymore. The reason: All the device drivers for the SoC network devices pull in GNRC as a static dependency.

#### Steps to reproduce the issue
Build `tests/lwip` for an ESP32 board, e.g.

```sh
BOARD=esp32-wroom-32 make -C tests/lwip
```

#### Expected results
`tests/lwip` should not pull in any GNRC dependencies.

#### Actual results
From the build output in current master:

```
"make" -C /data/riotbuild/riotbase/sys/net/gnrc
"make" -C /data/riotbuild/riotbase/sys/net/gnrc/netapi
"make" -C /data/riotbuild/riotbase/sys/net/gnrc/netif
"make" -C /data/riotbuild/riotbase/sys/net/gnrc/netif/hdr
"make" -C /data/riotbuild/riotbase/sys/net/gnrc/netreg
"make" -C /data/riotbuild/riotbase/sys/net/gnrc/pkt
"make" -C /data/riotbuild/riotbase/sys/net/gnrc/pktbuf
"make" -C /data/riotbuild/riotbase/sys/net/gnrc/pktbuf_static
```

#### Versions
Current master (50c2a2ca066a3725a4d79270e2559a6567893)

**Comment by miri64** [comment]
Ah proposed fix from my side would be to not make the GNRC dependencies static but to put it in their own sub-module, as we did e.g. with `xbee`, `cc110x`, etc as well.

**Comment by gschorcht** [comment]
Yeah, this is already fixed with the code deduplication PR #12955. Since PR#12955 will take some time before it is merged, I will provide a fix for the current master.

**Comment by gschorcht** [comment]
These are simple left overs from the time when I started with the complex building system and wanted to have a working network interface when I compile `tests/shell` with any network device.
